### PR TITLE
chore: implement audit log reader

### LINKED ISSUE #578: Add support for getting concatenated logs from audit log

### PR DESCRIPTION
**Pull request by DmitriyMV**
Implement audit log reader functionality and `audit-log` command in `omnictl`.

Closes #578

### CODE REVIEW

**Comment by DmitriyMV** [comment]
Output example:
```
~ > omnictl audit-log
{"event_type":"create","resource_type":"Identities.omni.sidero.dev","event_ts":1724267612220,"event_data":{"new_user":{"id":"3b470fcd-4170-420e-94f8-0ea03180ec35","email":"dmitry.matrenichev@siderolabs.com"},"session":{"user_agent":"Omni-Internal-Agent"}}}
{"event_type":"create","resource_type":"Users.omni.sidero.dev","event_ts":1724267612221,"event_data":{"new_user":{"role":"Admin","id":"3b470fcd-4170-420e-94f8-0ea03180ec35"},"session":{"user_agent":"Omni-Internal-Agent"}}}
{"event_type":"create","resource_type":"Identities.omni.sidero.dev","event_ts":1724267612221,"event_data":{"new_user":{"id":"e3015477-3428-4f76-bc6d-c29d24793e15","email":"test-user@siderolabs.com"},"session":{"user_agent":"Omni-Internal-Agent"}}}
{"event_type":"create","resource_type":"Users.omni.sidero.dev","event_ts":1724267612221,"event_data":{"new_user":{"role":"Admin","id":"e3015477-3428-4f76-bc6d-c29d24793e15"},"session":{"user_agent":"Omni-Internal-Agent"}}}
{"event_type":"create","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724267644257,"event_data":{"session":{"user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36","ip_address":"31.134.188.36","user_id":"3b470fcd-4170-420e-94f8-0ea03180ec35","role":"Admin","email":"dmitry.matrenichev@siderolabs.com","fingerprint":"e10f16f93905af3859be9726c05a8742143b6d9f","public_key_expiration":1724295844}}}
{"event_type":"update","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724267644924,"event_data":{"session":{"user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36","ip_address":"31.134.188.36","user_id":"3b470fcd-4170-420e-94f8-0ea03180ec35","role":"Admin","email":"dmitry.matrenichev@siderolabs.com","fingerprint":"e10f16f93905af3859be9726c05a8742143b6d9f","confirmation_type":"auth0","public_key_expiration":1724295844}}}
{"event_type":"create","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724267657466,"event_data":{"session":{"user_agent":"grpc-go/1.65.0","ip_address":"147.75.84.91","user_id":"3b470fcd-4170-420e-94f8-0ea03180ec35","role":"Admin","email":"dmitry.matrenichev@siderolabs.com","fingerprint":"d57c8abcd692392a945c36363b4295ff22b83a67","public_key_expiration":1724282057}}}
{"event_type":"update","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724267672326,"event_data":{"session":{"user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36","ip_address":"31.134.188.36","user_id":"3b470fcd-4170-420e-94f8-0ea03180ec35","role":"Admin","email":"dmitry.matrenichev@siderolabs.com","fingerprint":"d57c8abcd692392a945c36363b4295ff22b83a67","confirmation_type":"auth0","public_key_expiration":1724282057}}}
{"event_type":"destroy","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724282612242,"event_data":{"session":{"user_agent":"Omni-Internal-Agent","fingerprint":"d57c8abcd692392a945c36363b4295ff22b83a67"}}}
{"event_type":"destroy","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724296412240,"event_data":{"session":{"user_agent":"Omni-Internal-Agent","fingerprint":"e10f16f93905af3859be9726c05a8742143b6d9f"}}}
{"event_type":"create","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724323366470,"event_data":{"session":{"user_agent":"grpc-go/1.65.0","ip_address":"147.75.84.91","user_id":"3b470fcd-4170-420e-94f8-0ea03180ec35","role":"Admin","email":"dmitry.matrenichev@siderolabs.com","fingerprint":"efb02283198f3be9c14cab5df931b4c39f93a234","public_key_expiration":1724337766}}}
{"event_type":"update","resource_type":"PublicKeys.omni.sidero.dev","event_ts":1724323375810,"event_data":{"session":{"user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36","ip_address":"95.161.222.236","user_id":"3b470fcd-4170-420e-94f8-0ea03180ec35","role":"Admin","email":"dmitry.matrenichev@siderolabs.com","fingerprint":"efb02283198f3be9c14cab5df931b4c39f93a234","confirmation_type":"auth0","public_key_expiration":1724337766}}}
```

File: internal/backend/runtime/omni/audit/log_file.go [hunk 1]
```diff
@@ -126,3 +128,68 @@ func (l *LogFile) CleanupOldFiles(before time.Time) error {
...
+type multiCloser struct {
+	closers []io.Closer
+}
+
+func (m *multiCloser) Close() error {
+	var err error
+
+	for _, c := range m.closers {
+		if cerr := c.Close(); cerr != nil {
+			err = cerr
```
Comment (utkuozdemir) [inline internal/backend/runtime/omni/audit/log_file.go:190]: maybe use multierror instead?

**Review by utkuozdemir — APPROVED** [review]

File: internal/backend/grpc/management.go [hunk 2]
```diff
@@ -621,6 +623,36 @@ rolloutLoop:
 	return s.triggerManifestResync(ctx, requestContext)
 }
 
+// ReadAuditLog reads the audit log from the backend.
+func (s *managementServer) ReadAuditLog(_ *emptypb.Empty, srv grpc.ServerStreamingServer[management.ReadAuditLogResponse]) error {
+	rdr, err := s.auditor.ReadAuditLog()
```
Comment (Unix4ever) [inline internal/backend/grpc/management.go:628]: I think we should enforce Admin role here:
```
	if _, err := s.authCheckGRPC(src.Context(), auth.WithRole(role.Admin)); err != nil {
		return err
	}
```

File: internal/backend/runtime/omni/audit/log_file.go (hunk 1 above)
Comment (DmitriyMV) [reply internal/backend/runtime/omni/audit/log_file.go]: 👍

File: internal/backend/grpc/management.go (hunk 2 above)
Comment (DmitriyMV) [reply internal/backend/grpc/management.go:632]: 👍

**Pull request merged**